### PR TITLE
739 don't include Package.Dependencies element when serializing `PackageVersion[]`

### LINKED ISSUE #739: <Package.Dependencies> element is included when serializing `PackageVersion[]`
`<Package.Dependencies>` is a concept which only makes sense for test plans, but we include it in other instances, such as when serializing `PackageVersion[]`. This is annoying because it will add a dependency on the currently used version of OpenTAP in order to deserialize a data object, which will cause serializer errors when trying to deserialize that same string in an older version of OpenTAP.

Let's fix this :)

### PR DESCRIPTION
**Pull request by alnlarsen**
Update the PackageVersion serializer to disable writing `Package.Dependencies` when serializing either a single `PackageVersion`, or a single collection of `PackageVersion`s.

Closes #739

### CODE REVIEW

**Review by rmadsen-ks — CHANGES_REQUESTED** [review]
``var writePackageDependencies = Serializer.GetUsedTypes().Any(t => t.DescendsTo(typeof(TestPlan)));```

We don't want to only do this when the type is TestPlan.

If you dont want the behavior in some specific case, why not just add the option to disable the package dependency check?

File: Package/PackageDefinitionSerializerPlugin.cs [hunk 1]
```diff
@@ -101,7 +101,9 @@ public override bool Deserialize(XElement node, ITypeData t, Action<object> sett
         /// </summary>
         public override bool Serialize(XElement node, object obj, ITypeData expectedType)
         {
-            if (expectedType.IsA(typeof(PackageDef)) == false && expectedType.IsA(typeof(PackageIdentifier)) == false)
+            if (expectedType.IsA(typeof(PackageDef)) == false
```
Comment (rmadsen-ks) [inline Package/PackageDefinitionSerializerPlugin.cs:104]: can you modify the xml comment of this method to be more specific?

**Review by rmadsen-ks — CHANGES_REQUESTED** [review]
OK, I understand this now. Looks good, but I had a couple of suggestions.

**Comment by alnlarsen** [comment]
I have added some comments and improved the license element bug fix. Please take another look :)

**Review by rmadsen-ks — APPROVED** [review]
Looks good!

**Pull request merged**